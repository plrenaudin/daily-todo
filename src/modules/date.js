import { startOfWeek, format, addDays } from "date-fns";

/**
 * Returns current week of the day or today if no day parameters
 *
 * @param {string} day Day date in ISO format
 *
 * @returns {Date[]} Array containing the dates of the week
 */
const currentWeek = (day = isoFormatDate()) => {
  const firstDayOfWeek = startOfWeek(day, { weekStartsOn: 1 });
  return [...Array(7).keys()].map(i => addDays(firstDayOfWeek, i));
};

/**
 * Returns current week in ISO format of the day or today if no day parameters
 *
 * @param {string} day Day date in ISO format
 *
 * @returns {string[]} Array containing the dates of the week in ISO format
 */
const currentWeekISO = (day = isoFormatDate()) =>
  currentWeek(day).map(isoFormatDate);

/**
 * Returns the date ISO formated: YYYY-MM-DD
 *
 * @param {Date} date Date to format
 *
 * @returns string The date formatted
 */
const isoFormatDate = (date = new Date()) => format(date, "YYYY-MM-DD");

export { currentWeek, isoFormatDate, currentWeekISO };
