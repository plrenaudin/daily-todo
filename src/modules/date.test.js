import { currentWeek, isoFormatDate, currentWeekISO } from "./date";

it("gets the current week of the day", () => {
  expect(currentWeekISO("2019-02-06")).toMatchSnapshot();
});
it("gets the current week of the day", () => {
  expect(currentWeek("2019-02-06")).toMatchSnapshot();
});

it("formats the date to ISO", () => {
  expect(isoFormatDate(new Date(1549681335901))).toEqual("2019-02-08");
});
