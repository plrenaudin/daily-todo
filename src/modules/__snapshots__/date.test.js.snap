// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets the current week of the day 1`] = `
Array [
  "2019-01-31",
  "2019-02-01",
  "2019-02-02",
  "2019-02-03",
  "2019-02-04",
  "2019-02-05",
  "2019-02-06",
]
`;

exports[`gets the current week of the day 2`] = `
Array [
  2019-01-31T05:00:00.000Z,
  2019-02-01T05:00:00.000Z,
  2019-02-02T05:00:00.000Z,
  2019-02-03T05:00:00.000Z,
  2019-02-04T05:00:00.000Z,
  2019-02-05T05:00:00.000Z,
  2019-02-06T05:00:00.000Z,
]
`;
